### ISSUE TITLE
false-positive `xUnit1035` and `xUnit1036` errors

### ISSUE THREAD

**Issue by viceroypenguin** [xunit/xunit#2826]
#### Code:
```cs
public class TestClass
{
  public static IEnumerable<object[]> GetSequences(IEnumerable<int> seq) =>
    new object[] { seq };

  [Theory]
  [MemberData(nameof(GetSequences), new int[] { 1, 2, })]
  public void Test(IEnumerable<int> seq)
  {
    Assert.NotEmpty(seq);
  }
}
```

#### Expected: 
Code will compile with no errors, as it does with xunit 2.6.1

#### Actual:
```
error xUnit1035: The value is not convertible to the method parameter 'seq' of type 'System.Collections.Generic.IEnumerable<int>'. Use a compatible data value. (https://xunit.net/xunit.analyzers/rules/xUnit1035)
error xUnit1036: error xUnit1036: There is no matching method parameter for value: 2. Remove unused value(s), or add more parameter(s). (https://xunit.net/xunit.analyzers/rules/xUnit1036)
```

#### Additional Notes
Looks like this was introduced with xunit/xunit.analyzers#165, to fix #2189. However, it does not distinguish between an `object` array (which will be turned into separate parameters) and an `int` (or other type) array, which is passed as the first element of the `parameters` array, and is passed whole as a parameter to the target method. 

/CC: @JamesTerwilliger

**Comment by JamesTerwilliger** [comment]
@viceroypenguin My apologies. Looking into a fix now.

**Comment by JamesTerwilliger** [comment]
@bradwilson @viceroypenguin Candidate fix ready: https://github.com/xunit/xunit.analyzers/pull/170

**Comment by bradwilson** [comment]
Fixed by xunit/xunit.analzyers#170.

Available in `1.7.0-pre.11`.
